scripts/jenkins/ardana/ansible/roles/heat_stack/tasks/main.yml

```diff
@@ -22,11 +22,27 @@
   register: get_stack
   failed_when: false
 
+- name: Check if SES stack also exists
+  command: |
+    openstack --os-cloud {{ os_cloud }} stack show {{ ses_stack_name }} \
+      -c stack_status -f value
+  register: ses_stack
+  when: heat_action == 'delete'
+  failed_when: false
+
 - include_tasks: monitor.yml
   when:
     - get_stack.stdout != ''
     - heat_action == 'monitor'
 
+- include_tasks: delete.yml
```

Comment:
You actually need to run the whole role all over again to delete the SES stack. More than that, you actually need to run the whole `openstack-ardana-heat` job again (with a different heat stack name, of course). Simply calling `delete.yml` here may work if the ECP cloud behaves, but it will create a lot of problems if it doesn't. To give you one example of how this can go wrong, you don't "monitor"  the SES stack before you delete it, so you might end up deleting a SES stack that is already in a `delete_pending` state, and heat is just extremely sensitive to that kind of thing.

I recommend we do a complete overhaul of this logic in a following PR:
  - change the `heat_stack` role and the `openstack-ardana-heat` job to be more generic (e.g. name should be `openstack-heat` and accept a heat stack name instead of `ardana_env`)
  - move the role to a "common" location somewhere, where it can be included by both Ardana and SES
  - update the `openstack-ses` pipeline job to actually call `openstack-heat` to create the SES cluster, the same way that `openstack-ardana` does it

But all this is for the next PR...

Reply:
Update: this is also something that we might need to do for crowbar.

Reply:
> You actually need to run the whole role all over again to delete the SES stack. More than that, you actually need to run the whole `openstack-ardana-heat` job again (with a different heat stack name, of course). Simply calling `delete.yml` here may work if the ECP cloud behaves, but it will create a lot of problems if it doesn't. To give you one example of how this can go wrong, you don't "monitor" the SES stack before you delete it, so you might end up deleting a SES stack that is already in a `delete_pending` state, and heat is just extremely sensitive to that kind of thing.

I don't thinks this could happen, `delete.yml` also include `monitor.yml`.

Reply:
You're right. That was a bad example on my part, but the point is still valid. On top of that, you're "patching" a generic job and ansible role so that it handles a specific scenario, which is not a good way to go forward.